content/en/mobile_testing/mobile_tests/advanced_options.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Advanced Options for Mobile Test Steps
+kind: documentation
+description: Configure advanced options for Mobile Test Steps
+further_reading:
+- link: "https://www.datadoghq.com/blog/test-maintenance-best-practices/"
+  tag: "Blog"
+  text: "Best practices for maintaining end-to-end tests"
+- link: "/mobile_testing/mobile_tests//"
+  tag: "Documentation"
+  text: "Learn how to create mobile tests"
+- link: "/mobile_testing/mobile_tests/steps/"
+  tag: "Documentation"
+  text: "Learn how to create steps in mobile tests"
+- link: "/data_security/synthetics/"
+  tag: "Documentation"
+  text: "Learn about Synthetic Monitoring Data Security"
+---
+
+## Overview
+
+This page describes advanced options for Synthetic mobile tests. 
+
+## Locate an element
+
+### Datadog algorithm
+
+Flakiness is a pain point in end-to-end testing because tests occasionally fail when a frontend team implements changes, causing an identifier in your test to alert instead of an actual application issue.
+
+To prevent flaky tests, Datadog uses an algorithm that leverages a set of locators to target elements in mobile tests. A small change in the UI may modify an element (for example, moving it to another location). The mobile test automatically locates the element again based on points of reference that were not affected by the change. 
+
+When the test runs successfully, the mobile test recomputes (or "self heals") any broken locators with updated values, ensuring your tests do not break from simple UI updates and that your tests are automatically adapting to your mobile application's UI. 
+
+To ensure that your mobile test does not validate an unexpected change, use [assertions][1] in your test creation. Assertions allow you to define what is and what is not expected behavior associated with the test step journey. 
```

Comment:
```suggestion

To prevent flaky tests, Datadog uses an algorithm that leverages a set of locators to target elements in mobile app tests. A small change in the UI may modify an element (for example, moving it to another location). The mobile app test automatically locates the element again based on points of reference that were not affected by the change. 

When the test runs successfully, the mobile app test recomputes (or "self heals") any broken locators with updated values, ensuring your tests do not break from simple UI updates and that your tests are automatically adapting to your mobile application's UI. 

To ensure that your mobile app test does not validate an unexpected change, use [assertions][1] in your test creation. Assertions allow you to define what is and what is not expected behavior associated with the test step journey. 
```